<launch>

  <arg name="name" default=""/>
  <arg name="serial_number" default=""/>
  <arg name="conn_type" default="sta"/>
  <arg name="lib_log_level" default="ERROR"/>
  <arg name="video_resolution" default="360"/>
  <arg name="video_raw" default="1"/>
  <arg name="video_h264" default="false"/>
  <arg name="video_compressed" default="false"/>
  <arg name="audio_raw" default="false"/>
  <arg name="audio_opus" default="false"/>
  <arg name="chassis_rate" default="10"/>
  <!-- the rate [Hz] at which to gather the gimbal state -->
  <arg name="gimbal_rate" default="10"/>
  <arg name="sensor_adapter" default="false"/>
  <arg name="sensor_adapter_rate" default="10"/>
  <!-- wheter and where to diplay the battery state; one of
    - off: do not display
    - right: display on the right gimbal led
    - left: display on the left gimbal led
   -->
  <arg name="display_battery" default="off"/>
  <arg name="joint_state_rate" default="10.0"/>
  <arg name="camera_calibration_suffix" default=""/>
  <arg name="camera_calibration_directory" default="$(find-pkg-share robomaster_ros)/config"/>

  <include file="$(find-pkg-share robomaster_ros)/launch/main.launch">
    <arg name="gimbal" value="true"/>
    <arg name="blaster" value="true"/>
    <arg name="model" value="s1"/>
    <arg name="name" value="$(var name)"/>
    <arg name="serial_number" value="$(var serial_number)"/>
    <arg name="conn_type" value="$(var conn_type)"/>
    <arg name="lib_log_level" value="$(var lib_log_level)"/>
    <arg name="video_resolution" value="$(var video_resolution)"/>
    <arg name="video_raw" value="$(var video_raw)"/>
    <arg name="video_h264" value="$(var video_h264)"/>
    <arg name="video_compressed" value="$(var video_compressed)"/>
    <arg name="audio_raw" value="$(var audio_raw)"/>
    <arg name="audio_opus" value="$(var audio_opus)"/>
    <arg name="chassis_rate" value="$(var chassis_rate)"/>
    <arg name="joint_state_rate" value="$(var joint_state_rate)"/>
    <arg name="gimbal_rate" value="$(var gimbal_rate)"/>
    <arg name="sensor_adapter" value="$(var sensor_adapter)"/>
    <arg name="sensor_adapter_rate" value="$(var sensor_adapter_rate)"/>
    <arg name="camera_calibration_suffix" value="$(var camera_calibration_suffix)"/>
    <arg name="camera_calibration_directory" value="$(var camera_calibration_directory)"/>
  </include>

  <push-ros-namespace namespace="$(var name)"/>

  <node name="display_battery" pkg="robomaster_ros" exec="display_battery" output="screen"
     if="$(eval '\'$(var display_battery)\' != \'off\'')">
    <param name='side' value="$(var display_battery)"/>
  </node>
</launch>
